.circleci/fetch_deps.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+# used by all CI jobs
+pip install --user tox
+
+# only used by the benchmark job
+if [ "$CIRCLE_JOB" == "py36-benchmark" ]; then
```

Comment:
@carver I've put the condition into that `fetch_deps` file because all jobs need tox and so things seem a bit more DRY and organized to me.

Reply:
I was assuming you would remove the tox install from `fetch_deps` altogether so there would be duplication.

I slightly prefer the other approach. It keeps some circle-specific things out of the script, which could theoretically end up being moved/reused for other setup things. Even so, this is a tiny thing, and only an opinion, so please code it however you like. :)

Reply:
>It keeps some circle-specific things out of the script, which could theoretically end up being moved/reused for other setup things

True that! Changed it.